mlflow/projects/_project_spec.py

```diff
@@ -199,7 +200,8 @@ def _compute_path_value(self, user_param_value, storage_dir):
         basename = os.path.basename(user_param_value)
         dest_path = os.path.join(storage_dir, basename)
         if dest_path != user_param_value:
-            data.download_uri(uri=user_param_value, output_path=dest_path)
+            artifact_utils._download_artifact_from_uri(artifact_uri=user_param_value,
+                                                       output_path=dest_path)
```

Comment:
So to make sure I'm understanding this correctly, the fix here would be to just pass `storage_dir` to `_download_artifiact_from_uri` instead of `dest_path`, correct? Can we assume that `storage_dir` exists?

And yes, I can add that test. Do you have a recommendation on where to put it? The other tests for this handling are in `/projects/test_entry_point.py`, but they all mock away the actual call, so was wondering if there is a better place for it since I'll want to actually create the run and artifact.

Reply:
Specifying `storage_dir` rather than `dest_path` will write all artifacts to the top-level `storage_dir`. As a result, if two parameter values have the same base name, they may overwrite each other. We should instead create a subdirectory for each parameter value (see https://github.com/mlflow/mlflow/pull/2774#discussion_r419902847) and download artifacts to this subdirectory.

Regarding testing, I think we can introduce a test case to `projects/test_projects.py`, which currently contains some end-to-end test cases that invoke full projects. Let me know if that makes sense.

Reply:
Got it, makes sense.

Reply:
I'm still working on the new test for this, but wanted to push up the original changes for it, to make sure they look sane